app/main.py

```diff
@@ -1,3 +1,66 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int) -> Distance :
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return NotImplemented
+
+    def __iadd__(self, other: int) -> Distance:
```

Comment:
The `__iadd__` method's type annotation for `other` should be `Union[Distance, int, float]` to reflect that it can accept both `Distance` objects and numeric types.